### PR TITLE
build: export the EBPF_EXTERNAL_ENV_FLAGS variable

### PR DESCRIPTION
**Pull request by fntlnz**
Useful to configure BPF programs and related libraries from the outside.

This is necessary because when using this project as a dependency of another project with `add_subdirectory` environment variables are not passed down to the final commands because they happen at the execution of the specific generator instead of at generating phase.

### CODE REVIEW

File: cmake/modules/libelf.cmake [hunk 1]
```diff
@@ -14,7 +14,7 @@ ExternalProject_Add(
     libelf
     DOWNLOAD_COMMAND ""
     CONFIGURE_COMMAND ""
-    BUILD_COMMAND CC=${CMAKE_C_COMPILER} MFLAGS= MAKEFLAGS= WITH_TESTS=no WITH_BUILD_TOOLS=no WITH_ADDITIONAL_DOCUMENTATION=no WITH_PE=no WITH_ISA=no ${LIBELF_FLAGS} MAKEOBJDIR=${LIBELF_BUILD_DIR} bmake -C ${LIBELF_SRC} -e
```
Comment (nicholasberlin) [inline cmake/modules/libelf.cmake:17]: I notice that `${LIBELF_FLAGS}` is removed. Intentional?

Comment (fntlnz) [reply cmake/modules/libelf.cmake:17]: Yes it is intentional, it was replaced with this more generic one

**Review by nicholasberlin — APPROVED** [review]

File: GPL/HostIsolation/KprobeConnectHook/CMakeLists.txt [hunk 2]
```diff
@@ -34,7 +34,7 @@ set(KPROBECONNECTHOOK_CFLAGS
 
 set(KPROBECONNECTHOOK_BPF_OUTPUT ${TARGET_EBPF_DIR}/KprobeConnectHook.bpf.o)
 add_custom_command(OUTPUT ${KPROBECONNECTHOOK_BPF_OUTPUT}
-                  COMMAND ${CLANG} ${KPROBECONNECTHOOK_CFLAGS} -c ${CMAKE_CURRENT_SOURCE_DIR}/KprobeConnectHook.bpf.c -o ${KPROBECONNECTHOOK_BPF_OUTPUT}
+                  COMMAND ${EPBF_EXTERNAL_ENV_FLAGS} ${CLANG} ${KPROBECONNECTHOOK_CFLAGS} -c ${CMAKE_CURRENT_SOURCE_DIR}/KprobeConnectHook.bpf.c -o ${KPROBECONNECTHOOK_BPF_OUTPUT}
```
Comment (stanek-michal) [inline GPL/HostIsolation/KprobeConnectHook/CMakeLists.txt:37]: typo right? s/EPBF/EBPF

File: GPL/HostIsolation/TcFilter/CMakeLists.txt [hunk 3]
```diff
@@ -38,7 +38,7 @@ set(TCFILTER_CFLAGS
 
 set(TCFILTER_BPF_OUTPUT ${TARGET_EBPF_DIR}/TcFilter.bpf.o)
 add_custom_command(OUTPUT ${TCFILTER_BPF_OUTPUT}
-                  COMMAND ${CLANG} ${TCFILTER_CFLAGS} -emit-llvm -c ${CMAKE_CURRENT_SOURCE_DIR}/TcFilter.bpf.c -o - | ${LLC} -march=bpf -mcpu=v2 -filetype=obj -o  ${TCFILTER_BPF_OUTPUT}
+                  COMMAND ${EPBF_EXTERNAL_ENV_FLAGS} ${CLANG} ${TCFILTER_CFLAGS} -emit-llvm -c ${CMAKE_CURRENT_SOURCE_DIR}/TcFilter.bpf.c -o - | ${LLC} -march=bpf -mcpu=v2 -filetype=obj -o  ${TCFILTER_BPF_OUTPUT}
```
Comment (stanek-michal) [inline GPL/HostIsolation/TcFilter/CMakeLists.txt:41]: s/EPBF/EBPF

File: cmake/modules/libbpf.cmake [hunk 4]
```diff
@@ -18,7 +18,7 @@ ExternalProject_Add(
     libbpf
     DOWNLOAD_COMMAND ""
     CONFIGURE_COMMAND ""
-    BUILD_COMMAND CC=${CMAKE_C_COMPILER} make -C ${LIBBPF_SRC} BUILD_STATIC_ONLY=1 NO_PKG_CONFIG=1 OBJDIR=${LIBBPF_BUILD_DIR} INCLUDEDIR= LIBDIR= UAPIDIR= CFLAGS=${CMAKE_C_FLAGS} DESTDIR=${LIBBPF_TARGET_INCLUDE_DIR} install
+    BUILD_COMMAND CC=${CMAKE_C_COMPILER} ${EPBF_EXTERNAL_ENV_FLAGS} make -C ${LIBBPF_SRC} BUILD_STATIC_ONLY=1 NO_PKG_CONFIG=1 OBJDIR=${LIBBPF_BUILD_DIR} INCLUDEDIR= LIBDIR= UAPIDIR= CFLAGS=${CMAKE_C_FLAGS} DESTDIR=${LIBBPF_TARGET_INCLUDE_DIR} install
```
Comment (stanek-michal) [inline cmake/modules/libbpf.cmake:21]: s/EPBF/EBPF

File: cmake/modules/libelf.cmake [hunk 5]
```diff
@@ -14,7 +14,7 @@ ExternalProject_Add(
...
+    BUILD_COMMAND CC=${CMAKE_C_COMPILER} ${EPBF_EXTERNAL_ENV_FLAGS} MFLAGS= MAKEFLAGS= WITH_TESTS=no WITH_BUILD_TOOLS=no WITH_ADDITIONAL_DOCUMENTATION=no WITH_PE=no WITH_ISA=no MAKEOBJDIR=${LIBELF_BUILD_DIR} bmake -C ${LIBELF_SRC} -e
```
Comment (stanek-michal) [inline cmake/modules/libelf.cmake:17]: s/EPBF/EBPF

**Review by stanek-michal — APPROVED** [review]
apart from the typo, LGTM

File: cmake/modules/libbpf.cmake (hunk 4 above)
Comment (fntlnz) [reply cmake/modules/libbpf.cmake:21]: Good catch, thanks Michal!

**Pull request merged**